iree/compiler/Conversion/Common/test/linalg_bufferize.mlir

```diff
@@ -760,3 +760,72 @@ hal.interface @legacy_io attributes {sym_visibility = "private"} {
 //  CHECK-SAME:     strides = dense<[2, 3]> : vector<2xi64>
 //  CHECK-SAME:     ins(%[[INPUT]], %[[WINDOW]] : memref<1x4x6x1xf32>, memref<2x3xf32>)
 //  CHECK-SAME:    outs(%[[RET0]] : memref<1x2x2x1xf32>)
+
+// -----
+
+func @read_only_subtensor() {
+  %c0 = constant 0 : index
+  %c1 = constant 1 : index
+  %0 = hal.interface.binding.subspan @legacy_io::@wo2[%c0] : !flow.dispatch.tensor<writeonly:?x?xf32>
+  %1 = hal.interface.binding.subspan @legacy_io::@ro0[%c0] : !flow.dispatch.tensor<readonly:?x?xf32>
+  %2 = flow.dispatch.tensor.load %1 : !flow.dispatch.tensor<readonly:?x?xf32> -> tensor<?x?xf32>
+  %3 = hal.interface.binding.subspan @legacy_io::@ro1[%c0] : !flow.dispatch.tensor<readonly:?x?xf32>
+  %4 = flow.dispatch.tensor.load %3 : !flow.dispatch.tensor<readonly:?x?xf32> -> tensor<?x?xf32>
+  %workgroup_size_x = hal.interface.workgroup.size[0] : index
+  %workgroup_size_y = hal.interface.workgroup.size[1] : index
+  %workgroup_id_x = hal.interface.workgroup.id[0] : index
+  %workgroup_count_x = hal.interface.workgroup.count[0] : index
+  %workgroup_id_y = hal.interface.workgroup.id[1] : index
+  %workgroup_count_y = hal.interface.workgroup.count[1] : index
+  %5 = affine.apply affine_map<()[s0, s1] -> (s0 * s1)>()[%workgroup_id_y, %workgroup_size_y]
+  %6 = affine.apply affine_map<()[s0, s1] -> (s0 * s1)>()[%workgroup_count_y, %workgroup_size_y]
+  %dim0 = memref.dim %2, %c0 : tensor<?x?xf32>
+  %dim1 = memref.dim %2, %c1 : tensor<?x?xf32>
+  scf.for %arg0 = %5 to %dim0 step %6 {
```

Comment:
For bufferization tests, we don't need the loops? They complicate the test and come at a maintenance cost later when we want to update.

Reply:
Hmmm, I agree. Will leave this as is, and try to not do loops (i obviously borrowed it from a failing test).